src/context/cartContext/useCartServerCalls.js

```diff
@@ -0,0 +1,103 @@
+import { useEffect } from "react";
+import { useCart } from "./cart-context";
+import { useAlert } from "react-alert";
+import axios from "axios";
+
+function useCartServerCalls() {
+  const token = localStorage.getItem("ecommToken");
+  const { cartState, cartDispatch } = useCart();
+  const alert = useAlert();
+
+  useEffect(() => {
+    (async () => {
+      if (token) {
+        try {
+          const response = await axios.get("/api/user/cart", {
+            headers: { authorization: token },
+          });
+          if (response.status === 200 && response.data.cart.length !== 0) {
+            cartDispatch({
+              type: "ADD-TO-CART",
+              payload: response.data.cart,
+            });
+          }
+        } catch (error) {
+          alert.show("Internal Server Error", { type: "error" });
+        }
+      }
+    })();
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [token]);
+
+  const addToCart = async (product) => {
+    try {
+      if (cartState.items.find((item) => item._id === product._id)) {
+        alert.show("Already In Cart", { type: "info" });
+      } else {
+        const response = await axios.post(
+          "/api/user/cart",
+          { product },
+          {
+            headers: { authorization: token },
+          }
+        );
+        cartDispatch({
+          type: "ADD-TO-CART",
+          payload: response.data.cart,
+        });
+        alert.show("Added To Cart", { type: "success" });
+      }
+    } catch (err) {
+      alert.show("Server Error: Cannot be Added!", { type: "error" });
+    }
+  };
+
+  const deleteFromCart = async (productId) => {
+    try {
+      const response = await axios.delete(`/api/user/cart/${productId}`, {
```

Comment:
Can destructure the response here itself as 
```js
const {data:{cart:cartResponse}}=api call
```
For accessing nested data in objects